app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostUploadTaskFragment.java

```diff
@@ -166,7 +166,7 @@ protected Void doInBackground(Void... params) {
             fullSizeRef.putBytes(bytes).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
                 @Override
                 public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
-                    final Uri fullSizeUrl = taskSnapshot.getDownloadUrl();
+                    final Uri fullSizeUrl = fullSizeRef.getDownloadUrl().getResult();
```

Comment:
This will crash, right?  `getDownloadUrl` returns a task but calling `getResult()` synchronously won't work.